Fixes #30737 - Add VLAN tag to Ubuntu preseed if present

stefanlasiewski opened the pull request:
Add code to pass parameters from the Kernel commandline to preseed/debian-installer to add a VLAN if the VLAN tag is present on a network.

~I have an ERB question and will ask it in the code review.~ Resolved.

Commit:
Fixes #30492 - Fixed bug in Highly Available checkbox in ovirt cp

shiramax committed:
Fixes #29984 - Validate all ovirt volume attrs are string

Commit:
Fixes #30615 - Fix incorrect ipv6 error showing in host interfaces

domitea committed:
Fixes #30627 - mac address is not required for virtual resources

domitea committed:
Fixes #30627 - mac address is not required for virtual resources

domitea committed:
Fixes #30627 - mac address is not required for virtual resources

Commit:
Fixes #30638 - replace react-numeric-input with rc-number-input

Commit:
Fixes #30638 - replace react-numeric-input with rc-number-input

Commit:
Fixes #30638 - replace react-numeric-input with rc-number-input

sharvit committed:
Fixes #30684 - add password edit button to the http proxy form

It should allow users to send the form without editing the password

ehelms committed:
Fixes #29356: Use puma-plugin-systemd to provide systemd notify support

This points to a fork of puma-plugin-systemd maintained by theforeman
organization to ensure this works with Puma 4.

amirfefer committed:
Fixes #30596 - replace mount_react_component in config reports

adamruzicka committed:
Fixes #30730 - Implement systemd notify support for dynflow-sidekiq

stefanlasiewski committed:
Fixes #30737 - If VLAN is present, pass to debian-installer

if the VLAN tag is present on a network, pass parameters from the Kernel
commandline to preseed/debian-installer.

stefanlasiewski committed:
Fixes #30737 - If VLAN is present, pass to debian-installer

if the VLAN tag is present on a network, pass parameters from the Kernel
commandline to preseed/debian-installer.

stefanlasiewski committed:
Fixes #30737 - If VLAN is present, pass to debian-installer

if the VLAN tag is present on a network, pass parameters from the Kernel
commandline to preseed/debian-installer.

stefanlasiewski committed:
Fixes #30737 - If VLAN is present, pass to debian-installer

if the VLAN tag is present on a network, pass parameters from the Kernel
commandline to preseed/debian-installer.

stefanlasiewski committed:
Fixes #30737 - If VLAN is present, pass to debian-installer

if the VLAN tag is present on a network, pass parameters from the Kernel
commandline to preseed/debian-installer.

lzap commented on app/views/unattended/provisioning_templates/PXELinux/preseed_default_pxelinux.erb:
```diff
@@ -28,6 +28,10 @@ oses:
+  if @host.provision_interface.vlanid
+    vlan_opt = "netcfg/use_vlan=true netcfg/vlan_id=#{@host.provision_interface.vlanid}"
+    options << "#{vlan_opt}"
+  end
```
Can you make this just one line? Also squash everything into a single commit if possible. I haven't tested but it looks good.

stefanlasiewski replied:
Done. I assume you still want the `if` / `end` syntax so it's consistent with the other conditionals in this block.

MariaAga committed:
Fixes #30601 - replace mount_react_component in toasts

MariaAga committed:
Fixes #30601 - replace mount_react_component in toasts

MariaAga committed:
Fixes #30601 - replace mount_react_component in toasts

amirfefer committed:
Fixes #30603 - replace mount_react_component in templates helper

tbrisker committed:
Refs #27298 - drop unstable audit mailer test

This test has been failing randomly very often and three previous
attempts at fixing it have all failed.

tbrisker committed:
Fixes #30706 - Gracefully recover from CR connection error

If the connection fails to the compute resource due to networking
issues we should capture it nicely rather than causing a 500 page.

tbrisker committed:
Fixes #30706 - Gracefully recover from CR connection error

If the connection fails to the compute resource due to networking
issues we should capture it nicely rather than causing a 500 page.

tbrisker committed:
Fixes #30706 - Gracefully recover from CR connection error

If the connection fails to the compute resource due to networking
issues we should capture it nicely rather than causing a 500 page.

ekohl commented:
Would you mind shortening the commit title a bit? https://chris.beams.io/posts/git-commit/ has more info, but you can use paragraphs.

stefanlasiewski commented:
> Would you mind shortening the commit title a bit?

Done.

tbrisker commented:
@stefanlasiewski looks like there was some issue with your rebase, and it pulled in a bunch of other commits. could you `git rebase develop; git push -f` on your branch please to get it sorted?

ekohl commented:
Besides what @tbrisker mentioned, this looks good.

stefanlasiewski committed:
Merge branch '30737_add-vlan-support-preseed-pxelinux' of https://github.com/stefanlasiewski/foreman into 30737_add-vlan-support-preseed-pxelinux

stefanlasiewski commented:
> @stefanlasiewski looks like there was some issue with your rebase, and it pulled in a bunch of other commits. could you `git rebase develop; git push -f` on your branch please to get it sorted?

Oh no, I don't know what I did. I need to dig myself out of this a bit.  It needs more than `git rebase develop; git push -f`. 

I might not be able to get back to this until later this week.

(Sorry, I don't do this with git very often.)

stefanlasiewski committed:
Merge branch '30737_add-vlan-support-preseed-pxelinux' of https://github.com/stefanlasiewski/foreman into 30737_add-vlan-support-preseed-pxelinux

stefanlasiewski committed:
Merge branch 'develop' of https://github.com/theforeman/foreman into 30737_add-vlan-support-preseed-pxelinux

stefanlasiewski committed:
Merge branch '30737_add-vlan-support-preseed-pxelinux' of https://github.com/stefanlasiewski/foreman into 30737_add-vlan-support-preseed-pxelinux

tbrisker commented:
No worries @stefanlasiewski, I fixed it for you :) you'll need to use `git pull -f` though to get the updated version into your local branch if needed.

tbrisker commented:
ok to test

stefanlasiewski committed:
Merge branch '30737_add-vlan-support-preseed-pxelinux' of https://github.com/stefanlasiewski/foreman into 30737_add-vlan-support-preseed-pxelinux

tbrisker committed:
Refs #30737 - update snapshots

tbrisker commented on test/unit/foreman/renderer/snapshots/ProvisioningTemplate/PXELinux/Preseed default PXELinux.snap.txt:
```diff
@@ -15,7 +15,7 @@ DEFAULT linux
 LABEL linux
     KERNEL 
-    APPEND initrd= interface=auto url=http://foreman.some.host.fqdn/unattended/provision ramdisk_size=10800 root=/dev/rd/0 rw auto hostname=snapshothost console-setup/ask_detect=false console-setup/layout=USA console-setup/variant=USA keyboard-configuration/layoutcode=us localechooser/translation/warn-light=true localechooser/translation/warn-severe=true locale=en_US
+    APPEND initrd= interface=auto url=http://foreman.some.host.fqdn/unattended/provision ramdisk_size=10800 root=/dev/rd/0 rw auto hostname=snapshothost console-setup/ask_detect=false console-setup/layout=USA console-setup/variant=USA keyboard-configuration/layoutcode=us localechooser/translation/warn-light=true localechooser/translation/warn-severe=true netcfg/use_vlan=true netcfg/vlan_id= local ...
```
@stefanlasiewski I've updated the snapshot for you but this doesn't seem correct?

lzap replied:
This is not correct, I just merged @ekohl 's change which fixes this. Let me fix this.

ekohl commented on app/views/unattended/provisioning_templates/PXELinux/preseed_default_pxelinux.erb:
```diff
@@ -28,6 +28,9 @@ oses:
   else
     options << 'console-setup/ask_detect=false console-setup/layout=USA console-setup/variant=USA keyboard-configuration/layoutcode=us localechooser/translation/warn-light=true localechooser/translation/warn-severe=true'
   end
+  if @host.provision_interface.vlanid
```
Looks like there is a `has_vlanid?` helper that's also allowed in the jail
```suggestion
  if @host.provision_interface.has_vlanid?
```

lzap committed:
Refs #30737 - Add VLAN tag to Ubuntu preseed if present

Co-authored-by: Ewoud Kohl van Wijngaarden <ewoud@kohlvanwijngaarden.nl>

lzap committed:
Refs #30737 - Add VLAN tag to Ubuntu preseed if present

lzap commented:
I squashed and fixed tests in https://github.com/theforeman/foreman/pull/8201

stefanlasiewski committed:
Merge branch '30737_add-vlan-support-preseed-pxelinux' of https://github.com/stefanlasiewski/foreman into 30737_add-vlan-support-preseed-pxelinux

stefanlasiewski committed:
Revert "Merge branch '30737_add-vlan-support-preseed-pxelinux' of https://github.com/stefanlasiewski/foreman into 30737_add-vlan-support-preseed-pxelinux"

This reverts commit 167d959fac558e3d5cde14f281870ca185886d90, reversing
changes made to c1dabaffa121aa190e81d3f4e524350363313216.

lzap commented:
The change was already merged for Foreman 2.4 in https://github.com/theforeman/foreman/pull/8201/files

I am closing this PR, thank you!
